src/datatype/memory_bvecf32.rs

```diff
@@ -152,6 +153,31 @@ impl IntoDatum for BVecf32Output {
     }
 }
 
+impl FromDatum for BVecf32Output {
+    unsafe fn from_polymorphic_datum(datum: Datum, is_null: bool, _typoid: Oid) -> Option<Self> {
+        if is_null {
+            None
+        } else {
+            let ptr = NonNull::new(datum.cast_mut_ptr::<BVecf32Header>()).unwrap();
+            let q = unsafe {
+                NonNull::new(pgrx::pg_sys::pg_detoast_datum(ptr.cast().as_ptr()).cast()).unwrap()
+            };
+            Some(BVecf32Output(q))
+        }
+    }
+}
+
+unsafe impl UnboxDatum for BVecf32Output {
+    type As<'src> = BVecf32Output;
+    #[inline]
+    unsafe fn unbox<'src>(d: pgrx::Datum<'src>) -> Self::As<'src>
+    where
+        Self: 'src,
+    {
+        unsafe { BVecf32Output::from_datum(d.sans_lifetime(), false).unwrap() }
```

Comment:
Do not call this function. It's really hard to understand the ownership here. Copy the code above.